borg/crypto.pyx

```diff
@@ -1,11 +1,42 @@
-"""A thin OpenSSL wrapper"""
+"""An AEAD style OpenSSL wrapper
+
+Note: AES-GCM mode needs OpenSSL >= 1.0.1d due to bug fixes in OpenSSL.
```

Comment:
this changeset is still intentionally isolated (just crypto.pyx + test), integration will happen in 1.2.